### PR TITLE
feat: update Badge

### PR DESCRIPTION
**Pull request by liulinboyi**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
> 2. Resolve what problem.
> 3. Related issue link.

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

**Comment by tangjinzhou** [comment]
?

**Comment by liulinboyi** [comment]
> ?

最近有些事，这个没来得及兼容，现在兼容完了

**Comment by liulinboyi** [comment]
@tangjinzhou Please help review, thanks!

File: components/_util/props-util.js [hunk 1]
```diff
@@ -129,7 +129,7 @@ const getComponent = (instance, prop, options = instance, execute = true) => {
 };
 const getComponentFromProp = (instance, prop, options = instance, execute = true) => {
```
Comment (tangjinzhou) [inline components/_util/props-util.js:130]: 首先感谢 pr，但期望能够做一些基本的测试  这个方法已经不使用了  
使用 getComponent 替换 
另外麻烦同步修改下相应单测
npm test -- /components/badge/__tests__/index.test.js -- watch

Comment (liulinboyi) [reply components/_util/props-util.js:130]: 好的

Comment (liulinboyi) [reply components/_util/props-util.js:130]: > 首先感谢 pr，但期望能够做一些基本的测试 这个方法已经不使用了
> 使用 getComponent 替换
> 另外麻烦同步修改下相应单测
> npm test -- /components/badge/**tests**/index.test.js -- watch

好的

Comment (liulinboyi) [reply components/_util/props-util.js:130]: > 首先感谢 pr，但期望能够做一些基本的测试 这个方法已经不使用了
> 使用 getComponent 替换
> 另外麻烦同步修改下相应单测
> npm test -- /components/badge/**tests**/index.test.js -- watch

在写测试时，@vue/server-test-utils这个包没有Vue3的版本
```
import { render } from '@vue/server-test-utils';
```
image
提示版本不匹配

Comment (tangjinzhou) [reply components/_util/props-util.js:130]: 改成 mount

Comment (liulinboyi) [reply components/_util/props-util.js:130]: > 改成 mount

好的，我试一下。
```
import { mount } from '@vue/test-utils';
import Badge from '../index';
  it('badge dot not scaling count > 9', () => {
    const badge = {
      components: {Badge},
      template: '<Badge count="10" dot />',
    }
    const wrapper = mount(badge)
    expect(wrapper.findAll('.ant-card-multiple-words').length).toBe(0);
  });
```
"@vue/test-utils": "^2.0.0-alpha.6"

@tangjinzhou 是这样吗？谢谢！

Comment (tangjinzhou) [reply components/_util/props-util.js:130]: 如果能跑过就可以

Comment (liulinboyi) [reply components/_util/props-util.js:130]: > 如果能跑过就可以

好的。

**Comment by tangjinzhou** [comment]
关注下这个对应改动  https://github.com/vueComponent/ant-design-vue/commit/11d700ca9d2eaa835a81d2dbc6ede845ab2f85cc   @liulinboyi

**Pull request merged**